solutions/swb-app/src/dataSetRoutes.ts

```diff
@@ -151,11 +151,38 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
       const pageSize = toNumber(req.query.pageSize) || DEFAULT_API_PAGE_SIZE;
       const paginationToken = req.query.paginationToken?.toString();
 
+      if (pageSize < 1 || pageSize > MAX_API_PAGE_SIZE) {
+        throw Boom.badRequest(`Page size must be between 1 and ${MAX_API_PAGE_SIZE}`);
+      }
+
       const response = await dataSetService.listDataSets(authenticatedUser, pageSize, paginationToken);
       res.send(response);
     })
   );
 
+  // List project dataSets
+  router.get(
+    '/projects/:projectId/datasets',
+    wrapAsync(async (req: Request, res: Response) => {
+      const authenticatedUser = validateAndParse<AuthenticatedUser>(AuthenticatedUserParser, res.locals.user);
+      const pageSize = toNumber(req.query.pageSize) || DEFAULT_API_PAGE_SIZE;
+      const paginationToken = req.query.paginationToken?.toString();
+      const projectId = req.params.projectId;
+
+      if (pageSize < 1 || pageSize > MAX_API_PAGE_SIZE) {
```

Comment:
Should be idempotent, if they called a larger number than MAX, just floor it to the MAX

Reply:
I prefer erroring so that this doesn't lead to unexpected behavior.